### PR TITLE
Handle unfinished jobs after dead workers

### PR DESCRIPTION
**Pull request by pawbu**
When setting `mem_limit` in docker compose, single workers can be killed while others and the main app is OK and running. We need to check if a dead worker has not finished a job. If that is the case, we need to set the job to failed and clean up working directory.

### CODE REVIEW

File: job_executor/app.py [hunk 1]
```diff
@@ -62,75 +62,76 @@ def fix_interrupted_jobs():
...
+        if job.status == "importing":
+            rollback.rollback_manager_phase_import_job(
+                job.job_id, job_operation, job.parameters.target
+            )
             logger.info(
-                'Setting status to "queued" for '
-                f"interrupted job with id {job.job_id}"
+                f"{job.job_id}: Rolled back importing of job with "
+                f'operation "{job_operation}". Retrying from status '
+                '"built"'
```
Comment (haraldsvik) [inline job_executor/app.py:81]: A nitpick, Forgot I had pyright on and saw it complain about "built" not being a valid status however it is a valid based on our pydantic model.

Pyright wants to use enum members: JobStatus.BUILT. This provides better typesafety(typosafety) and if we are considering pyright as a typechecker in the future these are small fixes we can do along the way.

File: job_executor/app.py [hunk 2]
```diff
@@ -227,6 +228,11 @@ def main():
             built_jobs = job_dict["built_jobs"]
             queued_manager_jobs = job_dict["queued_manager_jobs"]
 
+            dead_workers = [
+                worker for worker in workers if not worker.is_alive()
+            ]
+            clean_up_after_dead_workers(dead_workers)
```
Comment (haraldsvik) [inline job_executor/app.py:234]: 💯

**Review by haraldsvik — DISMISSED** [review]
Looks good to me!

File: job_executor/app.py (hunk 1 above)
Comment (pawbu) [reply job_executor/app.py:81]: Yes, true. I see that Pylance is reporting this as well as some other problems. I suggest we fix those in a separate task (and we should standarize on a type checker across our projects).

Comment (DanielElisenberg) [reply job_executor/app.py:81]: agreed! (and pylance uses pyright under the hood so both of you getting the same errors makes sense)

**Comment by DanielElisenberg** [comment]
I like it! But we should rename StartupException as it can now be raised to the main loop through a rollback of an interrupted job ( Or at least raise it originally as a RollbackException or similar and then name it more specifically in the respective cases)

**Review by DanielElisenberg — APPROVED** [review]
🚀 :100:

**Pull request merged**